scripts/compile.sh

```diff
@@ -101,11 +102,13 @@ load_llvm() {
 }
 
 build_llvm() {
-    if ! [ -d "llvm-project" ]; then
+    local ext_dir="$PWD"
+    local llvm_dir="$ext_dir/llvm-project"
```

Comment:
what's the point of introducing the variable? this does not affect execution IMHO

Reply:
any variable may hold some invalid value at some point while constant is always a constant

Reply:
> what's the point of introducing the variable? this does not affect execution IMHO

To save the directory path before `cd`

Reply:
> any variable may hold some invalid value at some point while constant is always a constant

Sorry, didn't get the point. These local variables are used to temporary save the paths.

Reply:
just use the paths; I cannot see how adding variables adds a value to the script; it adds a few lines and a chance to have less transparent problems later

Reply:
With relative paths the following code fails, because lvm-project is a link in my env.

```shell
        cd ../llvm-project
        git apply ../mlir-extensions/build_tools/patches/*
```

Reply:
llvm-project and mlir-extensions are physically located in different folders:

```shell
ls -la externals/
...
llvm-project -> /home/andrey/work/llvm-project
mlir-extensions
```

When running `git apply ../mlir-extensions/build_tools/patches/*` from the `llvm-project` folder, the relative path is not valid.